crates/router/src/db/capture.rs

```diff
@@ -0,0 +1,403 @@
+        ) -> CustomResult<(Capture, storage::PaymentAttempt), errors::StorageError> {
+            self.validate_attempt_and_capture(&parent_attempt, &this.authorized_attempt_id)?;
+            let db_call = || async {
+                let conn = connection::pg_connection_write(self).await?;
+                let previous_capture_status = this.status;
+                let updated_capture = this
+                    .update_with_capture_id(&conn, capture)
+                    .await
+                    .map_err(Into::into)
+                    .into_report()?;
+                let attempt_update = if updated_capture.status != previous_capture_status {
+                    //if capture status is updated, lets update attempt status accordingly
+                    match updated_capture.status {
+                        enums::CaptureStatus::Charged => {
+                            let total_amount_captured =
+                                intent.amount_captured.unwrap_or_default() + updated_capture.amount;
+                            let authorized_amount = parent_attempt.amount;
+                            Some(
+                                storage::PaymentAttemptUpdate::MultipleCaptureResponseUpdate {
+                                    status: Some(if total_amount_captured < authorized_amount {
+                                        enums::AttemptStatus::PartialCharged
+                                    } else {
+                                        enums::AttemptStatus::Charged
+                                    }),
+                                    multiple_capture_count: None,
+                                },
+                            )
+                        }
+                        api_models::enums::CaptureStatus::Pending => Some(
+                            storage::PaymentAttemptUpdate::MultipleCaptureResponseUpdate {
+                                status: Some(enums::AttemptStatus::CaptureInitiated),
+                                multiple_capture_count: None,
+                            },
+                        ),
+                        //for rest of the cases, don't update payment_attempt
+                        api_models::enums::CaptureStatus::Started
+                        | api_models::enums::CaptureStatus::Failure => None,
+                    }
+                } else {
+                    None
+                };
+                let updated_attempt = match attempt_update {
+                    Some(payment_attempt_update) => {
+                        self.update_payment_attempt_with_attempt_id(
+                            parent_attempt,
+                            payment_attempt_update,
+                            storage_scheme,
+                        )
+                        .await?
+                    }
+                    None => parent_attempt,
+                };
+                Ok((updated_capture, updated_attempt))
+            };
```

Comment:
This business logic should be performed elsewhere, not in this function. This is for db operations only